crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -481,6 +481,44 @@ def upgrade_all_compute_nodes
         true
       end
 
+      def execute_scripts_and_wait_for_finish(nodes, script, seconds)
+        nodes.each do |node|
+          ssh_status = node.ssh_cmd(script).first
+          if ssh_status != 200
+            save_error_state("Failed to connect to node #{node.name}!")
+            raise "Execution of script #{script} has failed on node #{node.name}."
```

Comment:
I do not think it really matters. When the whole function is executed again, and scripts are run again on the nodes, they are not run again when there's already -ok file.